scripts/Dockerfile.build

```diff
@@ -0,0 +1,43 @@
+ARG VERSION
+FROM golang:1.16.5 as builder
+RUN mkdir -p /tmp/dd/datadog-agent
+
+# cache dependencies
+COPY ./scripts/.cache/datadog-agent /tmp/dd/datadog-agent
+WORKDIR /tmp/dd/datadog-agent
+RUN go mod graph \
+| grep -v winutil \
+| grep -v 'k8s.io/api' \
+| grep -v 'andybalholm' \
+| grep -v 'StackExchange/wmi' \
+| grep -v 'openshift/api' \
+| grep -v 'dd-trace-go.v1' \
+| grep -v 'iouring-go' \
+| grep -v 'lxn/walk' \
+| grep -v 'lxn/win' \
+| grep -v 'golang.org/genproto' \
+| awk '{if ($1 !~ "@") print $2}' | xargs go get
+RUN go get github.com/DataDog/gostackparse@v0.5.0
+RUN go mod tidy
+
+# copy source files (/tgz gets unzip automatically by Docker)
+ADD ./scripts/.src/datadog-agent.tgz /tmp/dd
+
+# build the extension
+WORKDIR /tmp/dd/datadog-agent/cmd/serverless
+# add the current version number to the tags package before compilation
+RUN go build -ldflags=LD_FLAGS="-s -w -X github.com/DataDog/datadog-agent/pkg/serverless/tags.currentExtensionVersion=${VERSION}" -tags serverless -o datadog-agent
+
+# zip the extension
+FROM ubuntu:latest as compresser
+RUN apt-get update
+RUN apt-get install -y zip
+RUN mkdir /extensions
+WORKDIR /extensions
+COPY --from=builder /tmp/dd/datadog-agent/cmd/serverless/datadog-agent /extensions/datadog-agent
+RUN  zip -r datadog_extension.zip /extensions
+
+# keep the smallest possible docker image
```

Comment:
Is there much point in doing this? Don't we just immediately copy the zip from the file?

Reply:
I think the goal here was to not include Go/Linux/the ZIP utility in the final image we produce. It might be worth clarifying this comment slightly.

Reply:
this allow us to create a super tiny container based on this image to run `docker create + docker cp` on it here : https://github.com/DataDog/datadog-lambda-extension/pull/18/files#diff-e3684a939c0316200e1a5d9554fd2700a4be757efd37109326b92183234115c4R38